README.md

```diff
@@ -1,169 +1,179 @@
 
-bc. <?hh
+```hack
 $post =
   <div class="post">
     <h2>{$post}</h2>
     <p><span>Hey there.</span></p>
     <a href={$like_link}>Like</a>
   </div>;
+```
 
-One advantage that XHP has over string construction is that it enforces correct markup structure at compile time. That is, the expression @$foo = <h1>Header</h2>;@ is not a valid expression, because you can not close an @h1@ tag with a @/h2@. When building large chunks of markup it can be difficult to be totally correct. With XHP the compiler now checks your work and will refuse to run until the markup is correct.
+One advantage that XHP has over string construction is that it enforces correct markup structure at compile time. That is, the expression `$foo = <h1>Header</h2>;` is not a valid expression, because you can not close an `h1` tag with a `/h2`. When building large chunks of markup it can be difficult to be totally correct. With XHP the compiler now checks your work and will refuse to run until the markup is correct.
 
-h1. Dynamic Structures
+## Dynamic Structures
 
-Sometimes it may be useful to create a bunch of elements and dynamically add them as children to an element. All XHP objects support the @appendChild@ method which behaves similarly to the same Javascript method. For example:
+Sometimes it may be useful to create a bunch of elements and dynamically add them as children to an element. All XHP objects support the `appendChild` method which behaves similarly to the same Javascript method. For example:
 
-bc. <?hh
+```hack
 $list = <ul />;
+$items = ...;
+
 foreach ($items as $item) {
   $list->appendChild(<li>{$item}</li>);
 }
+```
 
-In the code, @<ul />@ creates a ul with no children. Then we dynamically append children to it for each item in the @$items@ list.
+In the code, `<ul />` creates a ul with no children. Then we dynamically append children to it for each item in the `$items` list.
 
-h1. Escaping
+## Escaping
 
-An interesting feature of XHP is the idea of automatic escaping. In vanilla PHP if you want to render input from the user you must manually escape it. This practice is error-prone and has been proven over time to be an untenable solution. It increases code complexity and still leads to security vulnerabilities by careless programming. However, since XHP has context-specific about the page structure it can automatically escape data. The following two examples are identical, and both are "safe".
+An interesting feature of XHP is the idea of automatic escaping. In vanilla Hack if you want to render input from the user you must manually escape it. This practice is error-prone and has been proven over time to be an untenable solution. It increases code complexity and still leads to security vulnerabilities by careless programming. However, since XHP has context-specific about the page structure it can automatically escape data. The following two examples are identical, and both are "safe".
```

Comment:
Since XHP is a language feature of hack, I don't know if it makes sense to say "In vanilla Hack ..." (vs PHP where an optional extension was needed).

Perhaps "When using strings, ..."

Reply:
updated to : `If you want to render input from the user without using XHP, you must manually escape it....`